torrent/utilities.py

```diff
@@ -0,0 +1,53 @@
+from queue import Queue
+from threading import Thread
+from collections import OrderedDict
+import logging
+
+logger = logging.getLogger(__name__)
+
+#utilities
+def run_async(f, daemon=False):
+    def wrapped_f(q, *args, **kwargs):
+        ret = f(*args, **kwargs)
+        q.put(ret)
+    
+    def wrap(*args, **kwargs):
+        try:
+            if kwargs.get('recursive'):
+                return f(*args, **kwargs)
+        except KeyError:
+            pass
+        q = Queue() 
+        t = Thread(target=wrapped_f, args=(q, ) + args, kwargs=kwargs)
+        t.daemon = daemon
+        t.start()
+        t.result_queue = q
+        return t
+    return wrap
+
+def od_get_key(od, key, mandat=False):
```

Comment:
change mandat to something more understood, what does it mean???

Reply:
make it raise_on_failure or something

Reply:
fixed my brother